pysite/views/api/bot/user.py

```diff
@@ -56,10 +63,42 @@ def post(self, data):
 
         for item in all_oauth_data:
             if item["snowflake"] not in user_ids:
-                self.db.delete(self.oauth_table_name, item["id"], durability="soft")
-                self.db.delete(self.participants_table, item["id"], durability="soft")
-                oauth_deletions += 1
-                profile_deletions += 1
+                user_id = item["snowflake"]
+
+                oauth_deletions += self.db.delete(
+                    self.oauth_table_name, item["id"], durability="soft"
+                ).get("deleted", 0)
+                profile_deletions += self.db.delete(
+                    self.participants_table, user_id, durability="soft"
+                ).get("deleted", 0)
+
+                banned = False
+                responses = self.db.run(
+                    self.db.query(self.responses_table).filter({"snowflake": user_id}),
+                    coerce=list
+                )
+
+                for response in responses:
+                    jam = response["jam"]
+                    jam_obj = self.db.get(self.jams_table, jam)
+
+                    if jam_obj:
+                        if jam_obj["state"] in BANNABLE_STATES:
+                            banned = True
```

Comment:
maybe
```py
banned = jam_obj["state"] in BANNABLE_STATES
```
?

Reply:
Nope, users might be in more than one jam, so we have to at least do this check to break out of the loop.

Reply:
ah, okay